unhelpful error handling when bootstrap fails in maas newcloud

If you mistype a maas credential or don't type anything before hitting 'continue', you get an error message that just says 'error'. here's what was in the log:

```
154 Jul  7 18:15:56 brucewillis conjure-up/bundle-wordpress: [DEBUG] app.py:93 - Beta release found, checking minimum requirements.                                                                                                                                           
155 Jul  7 18:15:56 brucewillis conjure-up/bundle-wordpress: [DEBUG] download.py:130 - Checking remote URL: https://github.com/mikemccracken/bundle-wordpress/archive/master.zip                                                                                              
156 Jul  7 18:15:57 brucewillis conjure-up/bundle-wordpress: [DEBUG] download.py:82 - Extracting spell: bsdtar -xf /tmp/tmptam0mdm2/temp.zip -s'|[^/]*/||' -C /home/ubuntu/.cache/conjure-up/bundle-wordpress                                                                 
157 Jul  7 18:16:04 brucewillis conjure-up/bundle-wordpress: [DEBUG] gui.py:53 - Performing bootstrap: alvera maas/None                                                                                                                                                       
158 Jul  7 18:16:04 brucewillis conjure-up/bundle-wordpress: [DEBUG] juju.py:139 - bootstrap cmd: juju bootstrap alvera maas/None --upload-tools --config image-stream=daily --config enable-os-upgrade=false --bootstrap-series=xenial --credential margit                   
159 Jul  7 18:16:06 brucewillis conjure-up/bundle-wordpress: [DEBUG] gui.py:29 - handle bootstrap                                                                                                                                                                             
160 Jul  7 18:16:06 brucewillis conjure-up/bundle-wordpress: [ERROR] gui.py:36 - ERROR loading credentials: credentials.maas.alvera.maas-oauth: expected string, got nothing                                                                                                  
161 Jul  7 18:16:06 brucewillis conjure-up/bundle-wordpress: [ERROR] __init__.py:25 - Showing dialog for exception:#012NoneType                                                                                                                                               
```

--- a/conjureup/controllers/newcloud/gui.py
+++ b/conjureup/controllers/newcloud/gui.py
@@ -32,9 +32,15 @@ class NewCloudController:
             # bootstrap killed via user signal, we're quitting
             return
         if result.returncode > 0:
-            err = result.stderr.read().decode()
-            app.log.error(err)
-            return self.__handle_exception(Exception("error "))
+            pathbase = os.path.join(
+                app.config['spell-dir'],
+                '{}-bootstrap').format(app.current_controller)
+
+            with open(pathbase + ".err") as errf:
+                err = "\n".join(errf.readlines())
+                app.log.error(err)
+            e = Exception("Bootstrap error: {}".format(err))
+            return self.__handle_exception(e)
 
         utils.pollinate(app.session_id, 'J004')
         EventLoop.remove_alarms()
@@ -128,8 +134,6 @@ class NewCloudController:
         utils.pollinate(app.session_id, 'CA')
 
         self.__do_bootstrap(credential=credentials_key)
-
-        return controllers.use('deploy').render()
 
     def render(self, cloud):
         """ Render

